api/stream.go

```diff
@@ -162,8 +147,7 @@ func (s *StreamAPI) newSubscription(
 
 	subs := models.NewSubscription(callback(notifier, rpcSub))
 
-	rpcSub.ID = rpc.ID(subs.ID().String())
```

Comment:
I am a bit hesitant about removing this assignment here. If I remember correctly, this was added by Gregor, due to some issue with the event streaming API. But on the other hand, the CI is passing, so maybe it's not needed after all :man_shrugging:

Reply:
Thanks for the heads up, I'll take a look at the history, maybe there were some clues what specific problem this was solving.

Reply:
If you don't find anything, we can just merge it. It doesn't seem to break anything, and the E2E tests do exercise this part.

Reply:
@janezpodhostnik I did investigate this. What I found is:

```go
rpcSub.ID
```

has the format: `0x8b841deff1dbca0881c995ad77574d17`.
I think Gregor changed it to `uuid`, to match the format of the event streaming API,
e.g.: `69a20431-f601-43da-99f1-aa4bfd4d1bac`

But we're using entirely the subscription functionality from `Geth`, so there's no need to go with the `uuid` format.
I think we're all good here. It would make sense though, to log the proper `subscription-id`, e.g.
```go
l := logger.With().Str("subscription-id", string(rpcSub.ID)).Logger()
```

Reply:
addressed in [47ecbea](https://github.com/onflow/flow-evm-gateway/pull/602/commits/47ecbea56a0f2ea223d38b5af372099fd20feabd). Please check if that is what you had in mind.

Reply:
Nice, good idea adding both :+1: